File: Beecrowd/Beginner/1435 - 1985/1847 - Welcome to the Winter!/Python/1847_WelcometotheWinter!.py
temp1, temp2, temp3 = map(int, input().split())

if temp1 > temp2 and temp2 <= temp3:
    print(":)")
elif temp1 < temp2 and temp2 >= temp3:
    print(":(")
elif temp1 < temp2 and temp2 < temp3 and (temp3 - temp2) < (temp2 - temp1):
    print(":(")
elif temp1 < temp2 and temp2 < temp3 and (temp3 - temp2) >= (temp2 - temp1):
    print(":)")
elif temp1 > temp2 and temp2 > temp3 and (temp2 - temp3) < (temp1 - temp2):
    print(":)")
elif temp1 > temp2 and temp2 > temp3 and (temp2 - temp3) >= (temp1 - temp2):
    print(":(")
elif temp1 == temp2 and temp2 < temp3:
    print(":)")
elif temp1 == temp2 and temp2 > temp3:
    print(":(")
else:
    print(":(")

File: Beecrowd/Beginner/1435 - 1985/1847 - Welcome to the Winter!/C++/1847_WelcometotheWinter!.cpp
#include <bits/stdc++.h>

using namespace std;

int main() {
    int temp1, temp2, temp3;
    cin >> temp1 >> temp2 >> temp3;

    if (temp1 > temp2 && temp2 <= temp3)
        cout << ":)" << endl;
    else if (temp1 < temp2 && temp2 >= temp3)
        cout << ":(" << endl;
    else if (temp1 < temp2 && temp2 < temp3 && (temp3 - temp2) < (temp2 - temp1))
        cout << ":(" << endl;
    else if (temp1 < temp2 && temp2 < temp3 && (temp3 - temp2) >= (temp2 - temp1))
        cout << ":)" << endl;
    else if (temp1 > temp2 && temp2 > temp3 && (temp2 - temp3) < (temp1 - temp2))
        cout << ":)" << endl;
    else if (temp1 > temp2 && temp2 > temp3 && (temp2 - temp3) >= (temp1 - temp2))
        cout << ":(" << endl;
    else if (temp1 == temp2 && temp2 < temp3)
        cout << ":)" << endl;
    else if (temp1 == temp2 && temp2 > temp3)
        cout << ":(" << endl;
    else
        cout << ":(" << endl;

    return 0;
}